pulsar-broker/src/main/java/org/apache/pulsar/broker/web/PulsarWebResource.java

```diff
@@ -866,23 +866,42 @@ public void validateTenantOperation(String tenant, TenantOperation operation) {
         }
     }
 
-    public void validateNamespaceOperation(NamespaceName namespaceName, NamespaceOperation operation) {
+    public void validateNamespaceOperation(NamespaceName namespaceName, NamespaceOperation operation){
+        try {
+            validateNamespaceOperationAsync(namespaceName, operation).get();
+        } catch (InterruptedException e) {
+            throw new RestException(e);
+        } catch (ExecutionException e) {
+            Throwable cause = e.getCause();
+            if (cause instanceof WebApplicationException){
+                throw (WebApplicationException) cause;
+            } else {
+                throw new RestException(cause);
+            }
+        }
+    }
+
+
+    public CompletableFuture<Void> validateNamespaceOperationAsync(NamespaceName namespaceName,
+                                                              NamespaceOperation operation) {
         if (pulsar().getConfiguration().isAuthenticationEnabled()
             && pulsar().getBrokerService().isAuthorizationEnabled()) {
             if (!isClientAuthenticated(clientAppId())) {
                 throw new RestException(Status.FORBIDDEN, "Need to authenticate to perform the request");
```

Comment:
The exception won't be handled by `exceptionally` outside. No `CompletableFuture` is returned.

Reply:
fixed. PTAL :-)